include/exadg/operators/operator_base.cpp

```diff
@@ -549,8 +561,7 @@ OperatorBase<dim, Number, n_components>::initialize_block_diagonal_preconditione
   {
     typedef Elementwise::PreconditionerIdentity<dealii::VectorizedArray<Number>> IDENTITY;
 
-    elementwise_preconditioner =
-      std::make_shared<IDENTITY>(elementwise_operator->get_problem_size());
+    elementwise_preconditioner = std::make_shared<IDENTITY>(integrator->dofs_per_cell);
```

Comment:
To which extent is this change related to the present PR, could you explain this?

Reply:
This is a bugfix.

In this line we initialize the `elementwise_preconditioner` but the `elementwise_operator` is not initialized yet. This is why the value `elementwise_operator->get_problem_size()` is just the default value and not correct. 

I couldn't find any use case of the identity elementwise preconditioner, apart from the use case introduced in this PR. This is probably why, this bug was not found before.